### PR TITLE
Add validation when passing typed Textures to Bind Groups

### PR DESCRIPTION
**Pull request by reczkok**

### CODE REVIEW

File: packages/typegpu/src/tgpuBindGroupLayout.ts [hunk 1]
```diff
@@ -195,6 +204,91 @@ type StorageUsageForEntry<T extends TgpuLayoutStorage> = T extends {
...
+      format: T['storageTexture'];
+      dimension?: GetDimension<T['viewDimension']>;
+    }
+  > &
+    (
+      | Storage
+      | (Storage & Render)
+      | (Storage & Sampled)
+      | (Storage & Render & Sampled)
+    );
```
Comment (reczkok) [inline packages/typegpu/src/tgpuBindGroupLayout.ts]: 🤢
image

File: packages/typegpu/tests/bindGroupLayout.test.ts [hunk 2]
```diff
@@ -234,7 +239,7 @@ describe('TgpuBindGroup', () => {
     it('populates a simple layout with a raw buffer', ({ root }) => {
       const buffer = root.createBuffer(vec3f).$usage('uniform');
       const bindGroup = root.createBindGroup(layout, {
-        foo: root.unwrap(buffer),
+        foo: buffer,
```
Comment (iwoplaza) [inline packages/typegpu/tests/bindGroupLayout.test.ts]:
```suggestion
        foo: root.unwrap(buffer),
```

File: packages/typegpu/src/tgpuBindGroupLayout.ts [hunk 3]
```diff
@@ -195,6 +203,39 @@ type StorageUsageForEntry<T extends TgpuLayoutStorage> = T extends {
...
+    ? TgpuReadonlyTexture<
+        GetDimensionWithDefault<T['viewDimension']>,
+        TexelFormatToDataType[T['storageTexture']]
+      >
+    : TAccess extends 'writeonly'
+      ? TgpuWriteonlyTexture<
+          GetDimensionWithDefault<T['viewDimension']>,
+          TexelFormatToDataType[T['storageTexture']]
+        >
+      : never;
```
Comment (iwoplaza) [inline packages/typegpu/src/tgpuBindGroupLayout.ts]: 🗺️

**Review by iwoplaza — APPROVED** [review]
Stop right there, you're violating the law! 👨 ⚔️

**Pull request merged**